packages/sdk/identity/src/odis/identifier.ts

```diff
@@ -0,0 +1,204 @@
+  debug('Getting identifier pepper')
+
+  let seed: Buffer | undefined
+  if (blindingFactor) {
+    seed = Buffer.from(blindingFactor)
+  } else if (signer.authenticationMethod === AuthenticationMethod.ENCRYPTION_KEY) {
+    seed = Buffer.from((signer as EncryptionKeySigner).rawKey)
+  }
+
+  // Fallback to using Wasm version if not specified
+
+  if (!blsBlindingClient) {
+    debug('No BLSBlindingClient found, using WasmBlsBlindingClient')
+    blsBlindingClient = new WasmBlsBlindingClient(context.odisPubKey)
+  }
+
+  const base64BlindedMessage = await getBlindedIdentifier(
+    plaintextIdentifier,
+    identifierPrefix,
+    blsBlindingClient,
+    seed
+  )
+
+  const base64BlindSig = await getBlindedIdentifierSignature(
+    account,
+    signer,
+    context,
+    base64BlindedMessage,
+    clientVersion,
+    sessionID
+  )
+
+  return getObfuscatedIdentifierFromSignature(
+    plaintextIdentifier,
+    identifierPrefix,
+    base64BlindSig,
+    blsBlindingClient
+  )
+}
+
+/**
+ * Blinds the off-chain identifier in preparation for the ODIS request
+ * Caller should use the same blsBlindingClient instance for unblinding
+ */
+export async function getBlindedIdentifier(
+  identifier: string,
+  identifierPrefix: string,
+  blsBlindingClient: BlsBlindingClient,
+  seed?: Buffer
+): Promise<string> {
+  debug('Retrieving blinded message')
+  const base64Identifier =
+    identifierPrefix === IdentifierPrefix.PHONE_NUMBER
+      ? Buffer.from(identifier).toString('base64')
+      : Buffer.from(getPrefixedIdentifier(identifier, identifierPrefix)).toString('base64')
+  return blsBlindingClient.blindMessage(base64Identifier, seed)
+}
+
+/**
+ * Query ODIS for the blinded signature
+ * Response can be passed into getObfuscatedIdentifierFromSignature
+ * to retrieve the on-chain identifier
+ */
+export async function getBlindedIdentifierSignature(
+  account: string,
+  signer: AuthSigner,
+  context: ServiceContext,
+  base64BlindedMessage: string,
+  clientVersion?: string,
+  sessionID?: string,
+  keyVersion?: number,
+  endpoint?: CombinerEndpointPNP.LEGACY_PNP_SIGN | CombinerEndpointPNP.PNP_SIGN
+): Promise<string> {
+  const body: SignMessageRequest = {
+    account,
+    blindedQueryPhoneNumber: base64BlindedMessage,
+    version: clientVersion,
+    authenticationMethod: signer.authenticationMethod,
+    sessionID,
+  }
+
+  const response = await queryOdis(
+    body,
+    context,
+    endpoint ?? CombinerEndpointPNP.PNP_SIGN,
+    SignMessageResponseSchema,
+    {
+      [KEY_VERSION_HEADER]: keyVersion?.toString(),
+      Authorization: await getOdisPnpRequestAuth(body, signer),
+    }
+  )
+
+  if (!response.success) {
+    throw new Error(response.error)
+  }
+
+  return response.signature
+}
+
+/**
+ * Unblind the response and return the on-chain identifier
+ */
+export async function getObfuscatedIdentifierFromSignature(
+  plaintextIdentifier: string,
+  identifierType: string | IdentifierPrefix,
```

Comment:
should just be IdentifierPrefix, right?

Reply:
convo continued in https://github.com/celo-org/celo-monorepo/pull/9985#discussion_r1030759270